src/Sil/Component/Product/Model/Attribute.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2015-2018 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Product\Model;
+
+class Attribute
+{
+    /**
+     * Specialized name for this attribute.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * Value for attribute.
+     *
+     * @var mixed
+     */
+    protected $value;
+
+    /**
+     * The attribute for current value.
+     *
+     * @var AttributeType
+     */
+    protected $attributeType;
+
+    public function __construct(AttributeType $type, $value, $name = null)
+    {
+        $this->name = $name;
+        $this->value = $value;
+        $this->attributeType = $attributeType;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name ?: $this->getAttribute()->getName();
```

Comment:
should be `$this->getAttributeType()->getName()`

Reply:
Well spotted !